scripts/mbedtls_framework/code_wrapper/psa_wrapper.py

```diff
@@ -0,0 +1,286 @@
+        'psa_set_key_domain_parameters', # client-side function
+    ])
+
+    skipped_argument_types = frozenset([
+        # PAKE stuff: not implemented yet
+        'psa_crypto_driver_pake_inputs_t *',
+        'psa_pake_cipher_suite_t *',
+    ])
+
+    function_guards = {
+        'mbedtls_psa_register_se_key': 'defined(MBEDTLS_PSA_CRYPTO_SE_C)',
+        'mbedtls_psa_inject_entropy': 'defined(MBEDTLS_PSA_INJECT_ENTROPY)',
+        'mbedtls_psa_external_get_random': 'defined(MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG)',
+        'mbedtls_psa_platform_get_builtin_key': 'defined(MBEDTLS_PSA_CRYPTO_BUILTIN_KEYS)',
+        'psa_crypto_driver_pake_get_cipher_suite' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_password' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_password_len' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_peer' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_peer_len' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_user' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_user_len' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_abort' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_get_implicit_key' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_input' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_output' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_password_key' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_peer' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_role' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_user' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_setup' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+    }
+
+class PSAWrapper(c_wrapper_generator.Base):
+    """Generate a C source file containing wrapper functions for PSA Crypto API calls."""
+
+    _WRAPPER_NAME_PREFIX = 'mbedtls_test_wrap_'
+    _WRAPPER_NAME_SUFFIX = ''
+
+    _PSA_WRAPPER_INCLUDES = ['<psa/crypto.h>']
+    _DEFAULT_IN_HEADERS = ['crypto.h', 'crypto_extra.h']
+
+    def __init__(self,
+                 out_h_f: str,
+                 out_c_f: str,
+                 in_headers: Optional[List[str]] = None,
+                 config: PSAWrapperConfiguration = PSAWrapperConfiguration()) -> None:
+
+        super().__init__()
+        self.out_c_f = out_c_f
+        self.out_h_f = out_h_f
+
+        self.mbedtls_root = build_tree.guess_mbedtls_root()
+
+        self.read_config(config)
+        self.read_headers(in_headers)
+
+    def read_config(self, cfg: PSAWrapperConfiguration)-> None:
+        """Configure instance's parameters from a user provided config."""
+
+        self._CPP_GUARDS = PSAWrapper.parse_def_guards(cfg.cpp_guards)
+        self._SKIP_FUNCTIONS = cfg.skipped_functions
+        self._FUNCTION_GUARDS.update(cfg.function_guards)
+        self._NOT_IMPLEMENTED = cfg.skipped_argument_types
```

Comment:
As [previously discussed](https://github.com/Mbed-TLS/mbedtls-framework/pull/25#discussion_r1672399620), these are instance variables, not class variables, so the [naming convention](https://peps.python.org/pep-0008/#method-names-and-instance-variables) is lowercase. A leading underscore indicates that code outside the class should not access these fields (which I'm indifferent about).